.github/workflows/codeql-analysis.yml

```diff
@@ -15,6 +15,7 @@ on:
   push:
     branches: [ master ]
   pull_request:
+  merge_group:
```

Comment:
Does adding this change any semantics since the indented `branches` is now under `merge_group` and not `pull_request` ?

Reply:
Not actually sure, I originally thought it means `branches` would apply to both `pull_request` and `merge_group` because of this example: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue#triggering-merge-group-checks-with-github-actions. 

But after testing it out in a yaml parser that's not how it works. I think github just takes null as meaning it will apply to all pull requests. 

I'll update this.